apache/superset
# feat(native-filters): Adjust filter components for horizontal mode

### SUMMARY
This PR adjusts native filter components for horizontal filter bar.
Noteworthy changes:
1. Value filter:
   a) Introduce a `oneLine` mode to Select component - if enabled, only 1 tag will be displayed at once and the rest of selected tags will be collapsed to an overflow tag. When user opens the dropdown, only the overflow tag is displayed. All tags are always displayed in a single line.
   b) If tag is truncated, display a tooltip with full name
   c) Customize `maxTagPlaceholder`, so that the overflow tag doesn't display 3 dots at the end, which takes precious space 
       (e.g. instead of the default `+ 1 ...` we show `+1` when 1 item is overflowing)
3. Range filter:
   a) Adjust the top and bottom margin to make sure that the slider is centered in horizontal mode
   b) Decrease font weight and line height so that the marks fit well in the horizontal filter bar
4. Time range filter: add styling to make sure that the pill is centered in horizontal mode

Known issues:
1. The time range pill's width is dynamic depending on selected time range ("No filter" pill is smaller than "2022-01-01 : 2022-12-01" pill). However, the filter component itself always has a fixed width, which means that if the pill is short, there might be some empty space between the time range filter and the filter next to it. We are planning to replace it soon with a new component, stay tuned!

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<img width="1792" alt="Screenshot 2022-11-30 at 14 49 21" src="https://user-images.githubusercontent.com/15073128/204815039-5b059801-4ec0-4f9b-ba79-46ae40603a38.png">


### TESTING INSTRUCTIONS
1. Enable `HORIZONTAL_FILTER_BAR` ff
3. Set horizontal mode for the native filters bar
5. Add various types of filters
6. Make sure that nothing breaks

### ADDITIONAL INFORMATION




CC @codyml for review
